internal/featuredetection/feature_detection.go

```diff
@@ -50,9 +48,8 @@ type detector struct {
 }
 
 func NewDetector(httpClient *http.Client, host string) Detector {
-	cachedClient := api.NewCachedClient(httpClient, time.Hour*48)
 	return &detector{
-		httpClient: cachedClient,
+		httpClient: httpClient,
```

Comment:
Detector does not need to be aware of caching, let its callers be responsible for that.